src/azul/service/avro_pfb.py

```diff
@@ -143,7 +149,9 @@ def from_json(cls,
         # For files, document_id is not unique (because of related_files), but
         # uuid is.
         ids = [object_['uuid']] if name == 'files' else sorted(object_['document_id'])
-        return cls(id='_'.join((name, *ids)), name=name, object=object_)
+        id_ = str(uuid5(cls.namespace_uuid, '_'.join((name, *ids))))
+        assert len(id_) < 254, 'Terra requires IDs be shorter than 254 chars'
```

Comment:
nit. IMO its better to place assertions for instance attributes within `def __attrs_post_init__(self):` if for some reason this evolves, or a different class constructor is used you still get the benefits of the assertion.

This idea goes against the idea of asserting attributes before creating an instance w/ them in order to avoid malformed instances floating around in the runtime, but I've yet to think of a way that these malformed instances can exist without the runtime exiting due to the assertion/requirement firing off...

Reply:
This is a good suggestion, and inspired me to use attrs validators.